lib/glideinTracer.py

```diff
@@ -0,0 +1,107 @@
+# SPDX-FileCopyrightText: 2022 Fermi Research Alliance, LLC
+# SPDX-License-Identifier: Apache-2.0
+
+# GlideinWMS documentation for GlideFactoryLib.py, constructing
+# a Tracer and Trace class to give Glidein's unique TRACE_ID
+# and methods to send child spans or print the TRACE_ID of Glidein
+import os
+# fmt: off
+os.environ["OTEL_PROPAGATORS"] = "jaeger" 
+# fmt: on
+import time
+from opentelemetry import propagate, trace
+from opentelemetry.exporter.jaeger.thrift import JaegerExporter
+from opentelemetry.sdk.resources import Resource, SERVICE_NAME
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor
+from opentelemetry.trace.propagation.tracecontext import TraceContextTextMapPropagator
+
+
+
+# Global variables to establish tracing to Jaeger
+# ideally, these are read from the environment and have defaults (ex: server = "localhost", port = 6831)
+jaeger_collector_endpoint = "http://fermicloud296.fnal.gov:14268/api/traces?format=jaeger.thrift"
+
+
+class Tracer:
+    def __init__(self, collector_endpoint, jaeger_service_name="glidein"):
+        """Initializes a tracer with OpenTelemetry and Jaeger when operated in the GlideinWMS Factory
+
+        Args:
+            collector_endpoint (str): the http url for the collector endpoint (ex: "http://localhost:14268/api/traces?format=jaeger.thrift")
+            jaeger_service_name (str): the service to which the trace sends to through Jaeger
+        Variables:
+            GLIDEIN_TRACE_ID (hex): the parent trace_id of each submitted glidein instance
+            tracer: a tracer instance for each glidein to generate more spans
+            carrier = the injected SpanContext to be used to propogate child spans
+        """
+        self.collector_endpoint = collector_endpoint
+        self.jaeger_service_name = jaeger_service_name
+        self.GLIDEIN_TRACE_ID = None
+        self.tracer = None
+        self.carrier = None
```

Comment:
should the class variables be defined as private (starts with an `_` with getter and setter properties?